cbor/cbor.go

```diff
@@ -1515,7 +1515,7 @@ func collectFieldWeights(parents []int, i, upper int, field func(int) reflect.St
 
 	// Append to field list
 	return collectFieldWeights(parents, i+1, upper, field, append(fields, weightedField{
-		index:     append(parents, i),
+		index:     append(append([]int(nil), parents...), i),
```

Comment:
There are a few other places where we use parents. I suggest instead cloning the slice on line 1491:

```go
	// Duplicate parents slice, because it might be appended to
	parents = slices.Clone(parents)
```

Reply:
Updated it